src/main/java/org/jenkins/plugins/lockableresources/LockStepExecution.java

```diff
@@ -62,39 +62,43 @@ public boolean start() throws Exception {
     // determine if there are enough resources available to proceed
     List<LockableResource> available =
       LockableResourcesManager.get()
-        .checkResourcesAvailability(resourceHolderList, logger, null, step.skipIfLocked);
+        .checkResourcesAvailability(resourceHolderList, logger, null, step.skipIfLocked, step.resourceSelectStrategy);
     Run<?, ?> run = getContext().get(Run.class);
-    if (available == null
-      || !LockableResourcesManager.get()
+
+    if (available != null && LockableResourcesManager.get()
```

Comment:
Why we need change this logic and in lines above? When I see it correctly , it is the same but in 'negative' logic

Reply:
It removes a nesting layer, which makes it easier to read/understand in my opinion. But yes, the functionality is still the same.

Reply:
Can we make a deal. Please make only necessary changes here, and create other one PR like  code-refactoring.  There are many parts to be optimized and I fully agree with them.
Just because of reading the branch history. When somebody read the code and  try to find last changes will also read PR header. And this might leads to misunderstandings.

Reply:
thx